### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by goldsam** [dotnet/command-line-api#2285]
Is this project dead? There has only ever been beta prereleases and the last such was nearly a year and half ago. I want to be excited about a project backed by the .Net Foundation, but this project seems like an idea that was never followed through on. Is there any intent to make progress, or should I look at another option such as [`CommandLineParser`](https://www.nuget.org/packages/CommandLineParser)?

**Comment by goldsam** [comment]
I stand corrected. I looking at the examples on the main branch. The examples do work.

**Comment by jakenuts** [comment]
> Is this project dead? There has only ever been beta prereleases and the last such was nearly a year and half ago. The examples don't compile against the most recent nuget.org release. I want to be excited about a project backed by the .Net Foundation, but this project seems like an idea that was never followed through on. . Is there any intent to make progress, or should I look at another option such as [`CommandLineParser`](https://www.nuget.org/packages/CommandLineParser)?

The whole dotnet-suggest cli tab completion feature seems like it's gotta happen, though I'd prefer it in a low ceremony attribute based source-generated version.

**Comment by huoyaoyuan** [comment]
The productization of this repo is currently designed at dotnet/runtime repo (https://github.com/dotnet/runtime/issues/84177).

**Comment by goldsam** [comment]
@huoyaoyuan Are you talking about `Microsoft.Extensions.Configuration.CommandLine` which is a complete different library. I searched pretty thoroughly through the runtime repo, and I'm not seeing much there related to this repo/project. Could you please elaborate on what you mean by "productization of this repo"?

**Comment by huoyaoyuan** [comment]
> Are you talking about `Microsoft.Extensions.Configuration.CommandLine` (which is a completely different library)?

No. I'm even unaware of such a library. The linked issue (https://github.com/dotnet/runtime/issues/84177) is discussing API in `System.CommandLine` namespace.

> I searched pretty thoroughly through the runtime repo, and I'm not seeing much there related to this repo/project.

There is no code there currently, but only some issues for API review, tagged in `area-System.Console`.

> Could you please elaborate on what you mean by "productization of this repo"?

See #1882. In short,
*Once we release as part of the `System` namespace, we will consider the API locked and will work to avoid further breaking changes. Since we will live with that API, we need to take the time now to get it right.*

There are already pull requests tweaking namespace and type names like #2167.

**Comment by bhehe** [comment]
Following myself; I'm trying to advocate to use this library but the lack of a real release is a barrier for me at my employer.

**Comment by anthony-c-martin** [comment]
Agreed with the above. The other thing that makes this difficult is the lack of communication around an ETA.

We have been dying to use this library in https://github.com/Azure/bicep for 2 years. For this duration, this project has felt like it's "almost ready", so we built our own super basic solution, and didn't put energy into investigating alternatives (because "we'll be able to switch soon"). In retrospect, if we knew we'd be waiting for this long, we'd have just picked an alternative and moved on. It's hard to know that we won't be waiting another 2 years at this point.

**Comment by calacayir** [comment]
See my project here:
https://github.com/dotmake-build/command-line

DotMake.CommandLine is a library which provides declarative syntax for [System.CommandLine](https://github.com/dotnet/command-line-api) via attributes for easy, fast, strongly-typed (no reflection) usage. The library includes includes a source generator which automagically converts your classes to CLI commands and properties to CLI options or CLI arguments. Supports [trimming](https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-self-contained) and [AOT compilation](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net7%2Cwindows#limitations-of-native-aot-deployment) !

**Comment by lloydjatkinson** [comment]
Now that .NET 8 is out, can some time be given to this API? If not, I guess using `Microsoft.Extensions.Configuration.CommandLine` is a mostly OK option.

**Comment by mburumaxwell** [comment]
I believe `Microsoft.Extensions.Configuration.CommandLine` works for simple CLI applications where commands/sub-commands and other concepts are either not required or are simple enough to be written in the same app.
For other scenarios, either use the nightly builds (with breaking changes) or use [SpectreConsole](https://spectreconsole.net/) which has many more features and is stable.

**Comment by lloydjatkinson** [comment]
> I believe `Microsoft.Extensions.Configuration.CommandLine` works for simple CLI applications where commands/sub-commands and other concepts are either not required or are simple enough to be written in the same app. For other scenarios, either use the nightly builds (with breaking changes) or use [SpectreConsole](https://spectreconsole.net/) which has many more features and is stable.

You are right. It's strange that `System.CommandLine` appears to be being neglected. Entire new projects have been created and realised in the same time span, like Aspire...

**Comment by timker** [comment]
Hoping they find time to improve System.Commandline, but other options include (in no particular order):

- https://github.com/spectreconsole/spectre.console
- https://github.com/mayuki/Cocona
- https://github.com/commandlineparser/commandline

**Comment by BenjaminMichaelis** [comment]
> Hoping they find time to improve System.Commandline, but other options include (in no particular order):
> 

This project is actively being worked on; see https://github.com/dotnet/command-line-api/issues/2338

**Comment by aetos382** [comment]
Another advanced alternative: https://github.com/Cysharp/ConsoleAppFramework

**Comment by adamsitnik** [comment]
It's not dead for sure. We have released beta5 today: https://github.com/dotnet/command-line-api/issues/2576
